tools/azsdk-cli/Azure.Sdk.Tools.Cli.Tests/Services/Languages/JavaLanguageSpecificChecksTests.cs

```diff
@@ -150,13 +150,34 @@ private void SetupFailedSnippetUpdate()
         }
 
         /// <summary>
-        /// Sets up successful Maven javadoc command.
+        /// Sets up successful Maven linting command with all tools including javadoc.
+        /// This matches the fail-safe accumulation approach.
         /// </summary>
-        private void SetupSuccessfulJavadoc()
+        private void SetupSuccessfulMavenLinting()
         {
+            var pomPath = Path.Combine(JavaPackageDir, "pom.xml");
             MockProcessHelper.Setup(x => x.Run(It.Is<ProcessOptions>(p => 
```

Comment:
I wonder if at some point it would be worth creating a `MavenHelper` with `MavenOptions` that way the mocks can just look for `It.Is<MavenOptions>` and maybe filter on the args a bit more easily vs. having to enumerate all the args. Then we wouldn't have to check `mvn || cmd.exe` either.